{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 90452253,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T09:18:52.485Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 101142998,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T12:17:03.510Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 102100189,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T12:33:00.640Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 102203363,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T12:34:43.788Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 112552715,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T15:27:12.830Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 113220334,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-29T15:38:20.860Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 123741903,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-30T07:12:36.941Z' }
{ message: 'connect ECONNREFUSED 127.0.0.1:27017',
  reason:
   TopologyDescription {
     type: 'Single',
     setName: null,
     maxSetVersion: null,
     maxElectionId: null,
     servers:
      Map {
        'localhost:27017' => ServerDescription {
          address: 'localhost:27017',
          error:
           { Error: connect ECONNREFUSED 127.0.0.1:27017
               at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1107:14) name: 'MongoNetworkError' },
          roundTripTime: -1,
          lastUpdateTime: 124005934,
          lastWriteDate: null,
          opTime: null,
          type: 'Unknown',
          topologyVersion: undefined,
          minWireVersion: 0,
          maxWireVersion: 0,
          hosts: [],
          passives: [],
          arbiters: [],
          tags: [] } },
     stale: false,
     compatible: true,
     compatibilityError: null,
     logicalSessionTimeoutMinutes: null,
     heartbeatFrequencyMS: 10000,
     localThresholdMS: 15,
     commonWireVersion: null },
  level: 'error',
  timestamp: '2020-11-30T07:17:00.873Z' }
{ message:
   'Error: Provided address "undefined" is invalid, the capitalization checksum test failed, or its an indrect IBAN address which can\'t be converted.\n    at Method.inputAddressFormatter (/home/elliot/Desktop/exchange/server/node_modules/web3-core-helpers/lib/formatters.js:398:11)\n    at /home/elliot/Desktop/exchange/server/node_modules/web3-core-method/lib/index.js:129:38\n    at Array.map (<anonymous>)\n    at Method.formatInput (/home/elliot/Desktop/exchange/server/node_modules/web3-core-method/lib/index.js:127:32)\n    at Method.toPayload (/home/elliot/Desktop/exchange/server/node_modules/web3-core-method/lib/index.js:160:23)\n    at Eth.send [as getTransactionCount] (/home/elliot/Desktop/exchange/server/node_modules/web3-core-method/lib/index.js:498:61)\n    at Object.exports.getEtheriumNonce (/home/elliot/Desktop/exchange/server/api/walletApi/etheriuem.js:70:21)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:87:20\n    at Object.<anonymous> (/home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:47:42)\n    at step (/home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:33:23)\n    at Object.next (/home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:14:53)\n    at /home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:4:12)\n    at /home/elliot/Desktop/exchange/server/middlewares/tryCatch.js:41:47\n    at Layer.handle [as handle_request] (/home/elliot/Desktop/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n    at next (/home/elliot/Desktop/exchange/server/node_modules/express/lib/router/route.js:137:13)\n    at exports.validate (/home/elliot/Desktop/exchange/server/middlewares/validation.js:1866:9)\n    at Layer.handle [as handle_request] (/home/elliot/Desktop/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n    at next (/home/elliot/Desktop/exchange/server/node_modules/express/lib/router/route.js:137:13)\n    at middleware (/home/elliot/Desktop/exchange/server/node_modules/express-validator/src/middlewares/check.js:16:13)\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
<<<<<<< HEAD
  timestamp: '2020-11-23T13:49:08.562Z'
}
>>>>>>> 3c5d0ffa3c870a18254fc5d101dcee707be908b6
{
  message: 'MongoError: unknown operator: $or\n' +
    '    at MessageStream.messageHandler (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/connection.js:268:20)\n' +
    '    at MessageStream.emit (events.js:314:20)\n' +
    '    at processIncomingData (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n' +
    '    at MessageStream._write (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n' +
    '    at doWrite (_stream_writable.js:403:12)\n' +
    '    at writeOrBuffer (_stream_writable.js:387:5)\n' +
    '    at MessageStream.Writable.write (_stream_writable.js:318:11)\n' +
    '    at Socket.ondata (_stream_readable.js:719:22)\n' +
    '    at Socket.emit (events.js:314:20)\n' +
    '    at addChunk (_stream_readable.js:298:12)\n' +
    '    at readableAddChunk (_stream_readable.js:273:9)\n' +
    '    at Socket.Readable.push (_stream_readable.js:214:10)\n' +
    '    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)',
  level: 'error',
  timestamp: '2020-11-28T11:44:42.341Z'
}
{
  message: 'MongoError: unknown operator: $or\n' +
    '    at MessageStream.messageHandler (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/connection.js:268:20)\n' +
    '    at MessageStream.emit (events.js:314:20)\n' +
    '    at processIncomingData (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n' +
    '    at MessageStream._write (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n' +
    '    at doWrite (_stream_writable.js:403:12)\n' +
    '    at writeOrBuffer (_stream_writable.js:387:5)\n' +
    '    at MessageStream.Writable.write (_stream_writable.js:318:11)\n' +
    '    at Socket.ondata (_stream_readable.js:719:22)\n' +
    '    at Socket.emit (events.js:314:20)\n' +
    '    at addChunk (_stream_readable.js:298:12)\n' +
    '    at readableAddChunk (_stream_readable.js:273:9)\n' +
    '    at Socket.Readable.push (_stream_readable.js:214:10)\n' +
    '    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)',
  level: 'error',
  timestamp: '2020-11-28T11:45:26.928Z'
}
{
  message: 'MongoError: unknown operator: $or\n' +
    '    at MessageStream.messageHandler (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/connection.js:268:20)\n' +
    '    at MessageStream.emit (events.js:314:20)\n' +
    '    at processIncomingData (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n' +
    '    at MessageStream._write (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n' +
    '    at doWrite (_stream_writable.js:403:12)\n' +
    '    at writeOrBuffer (_stream_writable.js:387:5)\n' +
    '    at MessageStream.Writable.write (_stream_writable.js:318:11)\n' +
    '    at Socket.ondata (_stream_readable.js:719:22)\n' +
    '    at Socket.emit (events.js:314:20)\n' +
    '    at addChunk (_stream_readable.js:298:12)\n' +
    '    at readableAddChunk (_stream_readable.js:273:9)\n' +
    '    at Socket.Readable.push (_stream_readable.js:214:10)\n' +
    '    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)',
  level: 'error',
  timestamp: '2020-11-28T11:45:33.876Z'
}
{
  message: 'MongoError: unknown operator: $or\n' +
    '    at MessageStream.messageHandler (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/connection.js:268:20)\n' +
    '    at MessageStream.emit (events.js:314:20)\n' +
    '    at processIncomingData (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n' +
    '    at MessageStream._write (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n' +
    '    at doWrite (_stream_writable.js:403:12)\n' +
    '    at writeOrBuffer (_stream_writable.js:387:5)\n' +
    '    at MessageStream.Writable.write (_stream_writable.js:318:11)\n' +
    '    at Socket.ondata (_stream_readable.js:719:22)\n' +
    '    at Socket.emit (events.js:314:20)\n' +
    '    at addChunk (_stream_readable.js:298:12)\n' +
    '    at readableAddChunk (_stream_readable.js:273:9)\n' +
    '    at Socket.Readable.push (_stream_readable.js:214:10)\n' +
    '    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)',
  level: 'error',
  timestamp: '2020-11-28T11:46:58.186Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:55:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:06:55.088Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:56:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:08:40.813Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:56:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:10:01.928Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:56:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:11:31.876Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:56:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:12:31.176Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:57:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:13:17.314Z'
}
{
  message: 'CastError: Cast to ObjectId failed for value "undefined" at path "curGivenId" for model "AcceptedOffers"\n' +
    '    at model.Query.exec (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4380:21)\n' +
    '    at model.Query.Query.then (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/query.js:4472:15)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:57:14\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-28T12:14:15.506Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:387:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)',
  level: 'error',
  timestamp: '2020-11-29T08:20:50.852Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:387:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)',
  level: 'error',
  timestamp: '2020-11-29T08:21:33.740Z'
}
{
  message: 'ValidationError: User validation failed: wallet.0.commitment: Cast to Number failed for value "Infinity4250000000" at path "commitment"\n' +
    '    at model.Document.invalidate (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n' +
    '    at EmbeddedDocument.invalidate (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/types/embedded.js:291:29)\n' +
    '    at EmbeddedDocument.$set (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/document.js:1290:12)\n' +
    '    at EmbeddedDocument.set [as commitment] (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/helpers/document/compile.js:161:19)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:876:79\n' +
    '    at step (/home/hamid/projects/exchange/server/routes/user.js:44:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/routes/user.js:25:53)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:19:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/routes/user.js:15:12)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:854:76\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-30T07:23:57.485Z'
}
{
  message: 'ValidationError: User validation failed: wallet.0.commitment: Cast to Number failed for value "Infinity4250000000" at path "commitment"\n' +
    '    at model.Document.invalidate (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n' +
    '    at EmbeddedDocument.invalidate (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/types/embedded.js:291:29)\n' +
    '    at EmbeddedDocument.$set (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/document.js:1290:12)\n' +
    '    at EmbeddedDocument.set [as commitment] (/home/hamid/projects/exchange/server/node_modules/mongoose/lib/helpers/document/compile.js:161:19)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:876:79\n' +
    '    at step (/home/hamid/projects/exchange/server/routes/user.js:44:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/routes/user.js:25:53)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:19:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/routes/user.js:15:12)\n' +
    '    at /home/hamid/projects/exchange/server/routes/user.js:854:76\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:434:9)',
  level: 'error',
  timestamp: '2020-11-30T07:24:42.390Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:128:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T12:15:31.365Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:128:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T12:17:08.918Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:128:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T12:19:04.629Z'
}
=======
<<<<<<< HEAD
  timestamp: '2020-11-30T11:47:04.308Z' }
>>>>>>> 012bcd70d97c0eb31d1f5d15cae527fce24b6f74
=======
  timestamp: '2020-11-22T07:47:58.875Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.activeoffers index: userId_1 dup key: { userId: ObjectId(\'5fa2c4f34413a067b80199c4\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T07:57:04.860Z' }
{ message:
   'TypeError: offers.map(...).catch is not a function\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:154:76\n    at step (/home/elliot/Desktop/exchange/server/routes/wallet.js:33:23)\n    at Object.next (/home/elliot/Desktop/exchange/server/routes/wallet.js:14:53)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/elliot/Desktop/exchange/server/routes/wallet.js:4:12)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:84:93\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-11-22T08:26:51.445Z' }
{ message:
   'TypeError: Cannot read property \'value\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:115:85\n    at step (/home/elliot/Desktop/exchange/server/routes/wallet.js:33:23)\n    at Object.next (/home/elliot/Desktop/exchange/server/routes/wallet.js:14:53)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/elliot/Desktop/exchange/server/routes/wallet.js:4:12)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:94:107\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-11-22T08:34:12.895Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: offers.offerId_1 dup key: { offers.offerId: null }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)',
  level: 'error',
  timestamp: '2020-11-22T08:37:09.095Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: offers.offerId_1 dup key: { offers.offerId: null }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)',
  level: 'error',
  timestamp: '2020-11-22T08:38:23.070Z' }
{ message:
   'ParallelSaveError: Can\'t save() the same doc multiple times in parallel. Document: 5fa2987c287741a54c22aa2d\n    at model.Model.save (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:463:20)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:134:116\n    at step (/home/elliot/Desktop/exchange/server/routes/wallet.js:33:23)\n    at Object.next (/home/elliot/Desktop/exchange/server/routes/wallet.js:14:53)\n    at fulfilled (/home/elliot/Desktop/exchange/server/routes/wallet.js:5:58)\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-11-22T08:40:32.760Z' }
{ message:
   'ValidationError: AcceptedOffers validation failed: buyOrderId: Cast to ObjectId failed for value "b133a6cf-9aea-47af-bfae-0f4bdfd37f16" at path "buyOrderId"\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at model.$set (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:1290:12)\n    at model._handleIndex (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:1019:14)\n    at model.$set (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:960:22)\n    at model.Document (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:150:12)\n    at model.Model (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:106:12)\n    at new model (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:4685:15)\n    at toExecute.push.callback (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:3051:22)\n    at toExecute.forEach (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:3087:7)\n    at Array.forEach (<anonymous>)\n    at cb (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:3086:15)\n    at Promise (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/helpers/promiseOrCallback.js:31:5)\n    at new Promise (<anonymous>)\n    at promiseOrCallback (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/helpers/promiseOrCallback.js:30:10)\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/model.js:3021:10)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:129:139',
  level: 'error',
  timestamp: '2020-11-22T08:56:36.647Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)',
  level: 'error',
  timestamp: '2020-11-22T10:14:07.199Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)',
  level: 'error',
  timestamp: '2020-11-22T12:20:12.901Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)\n    at addChunk (_stream_readable.js:288:12)',
  level: 'error',
  timestamp: '2020-11-22T12:22:28.824Z' }
{ message:
   'ValidationError: AcceptedOffers validation failed: expiredDate: Path `expiredDate` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-22T12:33:35.711Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T12:36:24.139Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T12:45:59.215Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.activeoffers index: userId_1 dup key: { userId: ObjectId(\'5fa2987c287741a54c22aa2d\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T12:47:46.901Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-22T13:00:42.653Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T15:48:12.892Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-22T15:49:04.892Z' }
{ message: undefined,
  level: 'error',
  timestamp: '2020-11-22T15:50:51.473Z' }
{ message: undefined,
  level: 'error',
  timestamp: '2020-11-23T10:41:38.269Z' }
{ message: undefined,
  level: 'error',
  timestamp: '2020-11-23T10:44:32.884Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-23T10:46:47.213Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-23T10:47:49.628Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-23T10:49:58.054Z' }
{ message:
   'MongoError: E11000 duplicate key error collection: exchange.acceptedoffers index: acceptor_1 dup key: { acceptor: ObjectId(\'5fb2604b9d458a21da0af43b\') }\n    at Function.create (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/error.js:57:12)\n    at toError (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/utils.js:123:22)\n    at coll.s.topology.insert (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/operations/common_functions.js:265:39)\n    at handler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/topology.js:942:24)\n    at fn (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection_pool.js:350:13)\n    at handleOperationResult (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sdam/server.js:558:5)\n    at commandResponseHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/wireprotocol/command.js:115:25)\n    at MessageStream.messageHandler (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/connection.js:277:5)\n    at MessageStream.emit (events.js:198:13)\n    at processIncomingData (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:144:12)\n    at MessageStream._write (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/cmap/message_stream.js:42:5)\n    at doWrite (_stream_writable.js:415:12)\n    at writeOrBuffer (_stream_writable.js:399:5)\n    at MessageStream.Writable.write (_stream_writable.js:299:11)\n    at Socket.ondata (_stream_readable.js:710:20)\n    at Socket.emit (events.js:198:13)',
  level: 'error',
  timestamp: '2020-11-23T11:31:42.074Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-23T11:33:02.130Z' }
{ message: undefined,
  level: 'error',
  timestamp: '2020-11-23T11:34:00.750Z' }
{ message:
   'MongoError: Transaction already in progress\n    at ClientSession.startTransaction (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sessions.js:212:13)\n    at attemptTransaction (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sessions.js:350:11)\n    at ClientSession.withTransaction (/home/elliot/Desktop/exchange/server/node_modules/mongodb/lib/core/sessions.js:284:12)\n    at _loop_1 (/home/elliot/Desktop/exchange/server/routes/wallet.js:83:49)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:207:41\n    at step (/home/elliot/Desktop/exchange/server/routes/wallet.js:33:23)\n    at Object.next (/home/elliot/Desktop/exchange/server/routes/wallet.js:14:53)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/elliot/Desktop/exchange/server/routes/wallet.js:4:12)\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:71:86\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-11-23T11:44:45.990Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-23T11:59:50.674Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-23T14:01:17.776Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-23T14:11:30.408Z' }
{ message:
   'ValidationError: User validation failed: password: Path `password` is required.\n    at model.Document.invalidate (/home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2626:32)\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/document.js:2446:17\n    at /home/elliot/Desktop/exchange/server/node_modules/mongoose/lib/schematype.js:1225:9\n    at process._tickCallback (internal/process/next_tick.js:61:11)',
  level: 'error',
  timestamp: '2020-11-23T14:14:49.660Z' }
>>>>>>> 2d5b8ce82afa422db70b7d2b2dfa7a123d855817
=======
{
  message: 'The Sms service is not responding!',
  level: 'error',
  timestamp: '2020-11-23T13:27:38.907Z'
}
{
  message: 'The Sms service is not responding!',
  level: 'error',
  timestamp: '2020-11-23T13:49:08.562Z'
}
>>>>>>> 3c5d0ffa3c870a18254fc5d101dcee707be908b6
{
  message: 'RpcError: Expected type bool, got object\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T09:39:35.142Z'
}
{
  message: 'ReferenceError: unspentTx is not defined\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:120:17\n' +
    '    at Object.<anonymous> (/home/polychain3/exchange/server/middlewares/tryCatch.js:47:42)\n' +
    '    at step (/home/polychain3/exchange/server/middlewares/tryCatch.js:33:23)\n' +
    '    at Object.next (/home/polychain3/exchange/server/middlewares/tryCatch.js:14:53)\n' +
    '    at /home/polychain3/exchange/server/middlewares/tryCatch.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/polychain3/exchange/server/middlewares/tryCatch.js:4:12)\n' +
    '    at /home/polychain3/exchange/server/middlewares/tryCatch.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/polychain3/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/polychain3/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/polychain3/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/polychain3/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/polychain3/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at Function.process_params (/home/polychain3/exchange/server/node_modules/express/lib/router/index.js:335:12)\n' +
    '    at next (/home/polychain3/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/home/polychain3/exchange/server/node_modules/express/lib/router/index.js:174:3)',
  level: 'error',
  timestamp: '2020-11-25T10:57:09.572Z'
}
{
  message: 'RpcError: Invalid Bitcoin address: receiver\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T10:58:19.845Z'
}
{
  message: 'RpcError: Invalid Bitcoin address: receiver\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T10:59:26.649Z'
}
{
  message: 'RpcError: Invalid Bitcoin address: receiver\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T11:03:29.241Z'
}
{
  message: 'RpcError: Invalid Bitcoin address: receiver\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T11:04:50.676Z'
}
{
  message: 'RpcError: Invalid Bitcoin address: receiver\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T11:07:24.971Z'
}
{
  message: "RpcError: Data must be hexadecimal string (not 'Hi')\n" +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T11:21:58.764Z'
}
{
  message: "RpcError: Data must be hexadecimal string (not 'tb1qs0uffrs0n84q6v9krwe6ajx04j3mvxh0j96myl')\n" +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T11:24:34.961Z'
}
{
  message: 'TypeError: client.dumpprivkey is not a function\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:92:16\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T12:02:33.945Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-11-25T13:04:47.723Z'
}
{
  message: 'RpcError: Invalid parameter, key-value pair must contain exactly one key\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:20:40.389Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:23:21.051Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:28:19.715Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:32:58.151Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:37:09.723Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:47:43.484Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:48:28.082Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:50:01.470Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:53:11.598Z'
}
{
  message: 'RpcError: bad-txns-inputs-missingorspent\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T14:55:06.010Z'
}
{
  message: 'RpcError: bad-txns-vout-empty\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T15:41:01.898Z'
}
{
  message: 'RpcError: bad-txns-vout-empty\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T15:43:00.360Z'
}
{
  message: 'RpcError: min relay fee not met, 0 < 110\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-25T15:43:54.129Z'
}
{
  message: 'RpcError: JSON integer out of range\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T06:50:28.088Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:645:102\n' +
    '    at baseFindIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:802:11)\n' +
    '    at findIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:7280:14)\n' +
    '    at Function.find (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5079:21)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:645:25\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:08:45.933Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:649:102\n' +
    '    at baseFindIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:802:11)\n' +
    '    at findIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:7280:14)\n' +
    '    at Function.find (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5079:21)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:649:25\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:19:47.704Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:651:39\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:20:50.066Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:652:39\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:22:43.395Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:653:39\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:25:54.910Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:654:39\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-28T07:41:59.011Z'
}
{
  message: 'RpcError: Invalid amount\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T12:26:21.294Z'
}
{
  message: 'RpcError: Invalid amount\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T12:35:17.818Z'
}
{
  message: 'RpcError: Invalid amount\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:22:31.939Z'
}
{
  message: 'RpcError: Invalid amount\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:25:45.336Z'
}
{
  message: 'RpcError: bad-txns-in-belowout, value in (0.00003724) < value out (0.00190208)\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:31:26.053Z'
}
{
  message: 'RpcError: bad-txns-in-belowout, value in (0.00003724) < value out (0.00190208)\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:35:24.630Z'
}
{
  message: 'RpcError: Invalid parameter, missing vout key\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:38:11.150Z'
}
{
  message: 'RpcError: min relay fee not met, 100 < 141\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:41:06.896Z'
}
{
  message: 'RpcError: min relay fee not met, 130 < 141\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T13:43:51.288Z'
}
{
  message: 'Error: no auth mechanism defined\n' +
    '    at Auth.onRequest (/home/polychain3/exchange/server/node_modules/request/lib/auth.js:132:32)\n' +
    '    at Request.auth (/home/polychain3/exchange/server/node_modules/request/request.js:1341:14)\n' +
    '    at Request.init (/home/polychain3/exchange/server/node_modules/request/request.js:378:10)\n' +
    '    at new Request (/home/polychain3/exchange/server/node_modules/request/request.js:127:8)\n' +
    '    at request (/home/polychain3/exchange/server/node_modules/request/index.js:53:10)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:61:12\n' +
    '    at Object.apply (/home/polychain3/exchange/server/node_modules/@uphold/request-logger/src/index.js:27:19)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:100:12\n' +
    '    at /home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:41:9\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at Function.postAsync (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:40:38)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:48)\n' +
    '    at apply (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:474:27)\n' +
    '    at Client.wrapper [as getInfo] (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5337:16)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:127:12\n' +
    '    at Object.<anonymous> (/home/polychain3/exchange/server/middlewares/tryCatch.js:47:42)',
  level: 'error',
  timestamp: '2020-11-28T13:57:59.376Z'
}
{
  message: 'Error: no auth mechanism defined\n' +
    '    at Auth.onRequest (/home/polychain3/exchange/server/node_modules/request/lib/auth.js:132:32)\n' +
    '    at Request.auth (/home/polychain3/exchange/server/node_modules/request/request.js:1341:14)\n' +
    '    at Request.init (/home/polychain3/exchange/server/node_modules/request/request.js:378:10)\n' +
    '    at new Request (/home/polychain3/exchange/server/node_modules/request/request.js:127:8)\n' +
    '    at request (/home/polychain3/exchange/server/node_modules/request/index.js:53:10)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:61:12\n' +
    '    at Object.apply (/home/polychain3/exchange/server/node_modules/@uphold/request-logger/src/index.js:27:19)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:100:12\n' +
    '    at /home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:41:9\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at Function.postAsync (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:40:38)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:48)\n' +
    '    at apply (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:476:27)\n' +
    '    at Client.wrapper [as createRawTransaction] (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5337:16)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:134:12\n' +
    '    at Object.<anonymous> (/home/polychain3/exchange/server/middlewares/tryCatch.js:47:42)',
  level: 'error',
  timestamp: '2020-11-28T14:03:55.538Z'
}
{
  message: 'Error: no auth mechanism defined\n' +
    '    at Auth.onRequest (/home/polychain3/exchange/server/node_modules/request/lib/auth.js:132:32)\n' +
    '    at Request.auth (/home/polychain3/exchange/server/node_modules/request/request.js:1341:14)\n' +
    '    at Request.init (/home/polychain3/exchange/server/node_modules/request/request.js:378:10)\n' +
    '    at new Request (/home/polychain3/exchange/server/node_modules/request/request.js:127:8)\n' +
    '    at request (/home/polychain3/exchange/server/node_modules/request/index.js:53:10)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:61:12\n' +
    '    at Object.apply (/home/polychain3/exchange/server/node_modules/@uphold/request-logger/src/index.js:27:19)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:100:12\n' +
    '    at /home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:41:9\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at Function.postAsync (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:40:38)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:48)\n' +
    '    at apply (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:476:27)\n' +
    '    at Client.wrapper [as createRawTransaction] (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5337:16)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:134:12\n' +
    '    at Object.<anonymous> (/home/polychain3/exchange/server/middlewares/tryCatch.js:47:42)',
  level: 'error',
  timestamp: '2020-11-28T14:07:27.188Z'
}
{
  message: 'Error: no auth mechanism defined\n' +
    '    at Auth.onRequest (/home/polychain3/exchange/server/node_modules/request/lib/auth.js:132:32)\n' +
    '    at Request.auth (/home/polychain3/exchange/server/node_modules/request/request.js:1341:14)\n' +
    '    at Request.init (/home/polychain3/exchange/server/node_modules/request/request.js:378:10)\n' +
    '    at new Request (/home/polychain3/exchange/server/node_modules/request/request.js:127:8)\n' +
    '    at request (/home/polychain3/exchange/server/node_modules/request/index.js:53:10)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:61:12\n' +
    '    at Object.apply (/home/polychain3/exchange/server/node_modules/@uphold/request-logger/src/index.js:27:19)\n' +
    '    at /home/polychain3/exchange/server/node_modules/request/index.js:100:12\n' +
    '    at /home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:41:9\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at Function.postAsync (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:40:38)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:48)\n' +
    '    at apply (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:476:27)\n' +
    '    at Client.wrapper [as createRawTransaction] (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5337:16)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:134:12\n' +
    '    at Object.<anonymous> (/home/polychain3/exchange/server/middlewares/tryCatch.js:47:42)',
  level: 'error',
  timestamp: '2020-11-28T14:12:26.042Z'
}
{
  message: 'RpcError: Unauthorized\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:73:13)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T14:18:12.249Z'
}
{
  message: 'RpcError: Invalid parameter, duplicated address: tb1qqera5t2vuuhghtt8z92w3jcx47pu9xmfh4myy6\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T15:37:55.311Z'
}
{
  message: 'RpcError: min relay fee not met, 110 < 141\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T15:41:23.651Z'
}
{
  message: 'RpcError: min relay fee not met, 120 < 141\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T15:41:43.793Z'
}
{
  message: 'RpcError: min relay fee not met, 140 < 141\n' +
    '    at getRpcResult (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:39:11)\n' +
    '    at Parser.rpc (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/parser.js:82:14)\n' +
    '    at Client.command (/home/polychain3/exchange/server/node_modules/bitcoin-core/dist/src/index.js:169:24)\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-28T15:41:58.695Z'
}
{
  message: 'connection timed out',
  reason: TopologyDescription {
    type: 'Single',
    setName: null,
    maxSetVersion: null,
    maxElectionId: null,
    servers: Map {
      'localhost:27017' => ServerDescription {
        address: 'localhost:27017',
        error: MongoNetworkTimeoutError: connection timed out
            at connectionFailureError (/home/polychain3/exchange/server/node_modules/mongodb/lib/core/connection/connect.js:342:14)
            at Socket.<anonymous> (/home/polychain3/exchange/server/node_modules/mongodb/lib/core/connection/connect.js:310:16)
            at Object.onceWrapper (events.js:421:28)
            at Socket.emit (events.js:315:20)
            at Socket._onTimeout (net.js:482:8)
            at listOnTimeout (internal/timers.js:549:17)
            at processTimers (internal/timers.js:492:7),
        roundTripTime: -1,
        lastUpdateTime: 1214368,
        lastWriteDate: null,
        opTime: null,
        type: 'Unknown',
        topologyVersion: undefined,
        minWireVersion: 0,
        maxWireVersion: 0,
        hosts: [],
        passives: [],
        arbiters: [],
        tags: []
      }
    },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: null
  },
  level: 'error',
  timestamp: '2020-11-30T06:05:14.572Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:743:104\n' +
    '    at baseFindIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:802:11)\n' +
    '    at findIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:7280:14)\n' +
    '    at Function.find (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5079:21)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:743:27\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-30T07:28:46.091Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:743:104\n' +
    '    at baseFindIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:802:11)\n' +
    '    at findIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:7280:14)\n' +
    '    at Function.find (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5079:21)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:743:27\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-30T07:31:43.854Z'
}
{
  message: "TypeError: Cannot read property 'toString' of undefined\n" +
    '    at /home/polychain3/exchange/server/routes/wallet.js:746:104\n' +
    '    at baseFindIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:802:11)\n' +
    '    at findIndex (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:7280:14)\n' +
    '    at Function.find (/home/polychain3/exchange/server/node_modules/lodash/lodash.js:5079:21)\n' +
    '    at /home/polychain3/exchange/server/routes/wallet.js:746:27\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
    '    at runNextTicks (internal/process/task_queues.js:66:3)\n' +
    '    at processImmediate (internal/timers.js:429:9)',
  level: 'error',
  timestamp: '2020-11-30T07:36:17.690Z'
}
>>>>>>> 6c706ecec75a869c24841727e7d1325d9306c994
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T13:42:01.738Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T13:44:46.170Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T13:46:43.381Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T13:51:06.333Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:117:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T15:01:01.200Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:117:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T15:04:37.812Z'
}
{
  message: "TypeError: Cannot read property 'length' of undefined\n" +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:117:108\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T15:14:28.319Z'
}
{
  message: 'ReferenceError: offerIds is not defined\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:118:143\n' +
    '    at step (/home/hamid/projects/exchange/server/routes/wallet.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/routes/wallet.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/routes/wallet.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:124\n' +
    '    at attemptTransaction (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/core/sessions.js:354:15)\n' +
    '    at ClientSession.withTransaction (/home/hamid/projects/exchange/server/node_modules/mongodb/lib/core/sessions.js:284:12)\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:116:87\n' +
    '    at step (/home/hamid/projects/exchange/server/routes/wallet.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/routes/wallet.js:14:53)\n' +
    '    at fulfilled (/home/hamid/projects/exchange/server/routes/wallet.js:5:58)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T15:20:20.115Z'
}
{
  message: 'Error\n' +
    '    at new myError (/home/hamid/projects/exchange/server/api/myError.js:20:28)\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:131:93\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-11-30T15:23:23.560Z'
}
{
  message: 'Error\n' +
    '    at new myError (/home/hamid/projects/exchange/server/api/myError.js:20:28)\n' +
    '    at /home/hamid/projects/exchange/server/routes/wallet.js:132:93\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:97:5)',
  level: 'error',
  timestamp: '2020-12-01T05:42:16.184Z'
}
<<<<<<< HEAD
=======
>>>>>>> 3c5d0ffa3c870a18254fc5d101dcee707be908b6
{
  message: 'TypeError: TRONTransferTo_1.TRONTransferTo is not a function\n' +
    '    at /Users/Kwin/Documents/exchange/server/routes/user.js:83:22\n' +
    '    at Object.<anonymous> (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:47:42)\n' +
    '    at step (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:33:23)\n' +
    '    at Object.next (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:14:53)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:4:12)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at Function.process_params (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:335:12)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:174:3)',
  level: 'error',
  timestamp: '2020-11-30T14:03:03.338Z'
}
{
  message: 'TypeError: TRONTransferTo_1.TRONTransferTo is not a function\n' +
    '    at /Users/Kwin/Documents/exchange/server/routes/user.js:83:22\n' +
    '    at Object.<anonymous> (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:47:42)\n' +
    '    at step (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:33:23)\n' +
    '    at Object.next (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:14:53)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:4:12)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at Function.process_params (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:335:12)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:174:3)',
  level: 'error',
  timestamp: '2020-11-30T14:05:35.733Z'
}
{
  message: 'TypeError: TRONTransferTo_1.TRONTransferTo is not a function\n' +
    '    at /Users/Kwin/Documents/exchange/server/routes/user.js:83:22\n' +
    '    at Object.<anonymous> (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:47:42)\n' +
    '    at step (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:33:23)\n' +
    '    at Object.next (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:14:53)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:4:12)\n' +
    '    at /Users/Kwin/Documents/exchange/server/middlewares/tryCatch.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at Function.process_params (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:335:12)\n' +
    '    at next (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/Users/Kwin/Documents/exchange/server/node_modules/express/lib/router/index.js:174:3)',
  level: 'error',
  timestamp: '2020-11-30T14:05:37.460Z'
}
>>>>>>> 9b65fa157d7861bf4587956d1ac154d3f7ad8108
{ message:
   'Error: Invalid JSON RPC response: ""\n    at Object.InvalidResponse (/home/elliot/Desktop/exchange/server/node_modules/web3-core-helpers/lib/errors.js:43:16)\n    at XMLHttpRequest.request.onreadystatechange (/home/elliot/Desktop/exchange/server/node_modules/web3-providers-http/lib/index.js:95:32)\n    at XMLHttpRequestEventTarget.dispatchEvent (/home/elliot/Desktop/exchange/server/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:22)\n    at XMLHttpRequest._setReadyState (/home/elliot/Desktop/exchange/server/node_modules/xhr2-cookies/dist/xml-http-request.js:208:14)\n    at XMLHttpRequest._onHttpRequestError (/home/elliot/Desktop/exchange/server/node_modules/xhr2-cookies/dist/xml-http-request.js:349:14)\n    at ClientRequest.<anonymous> (/home/elliot/Desktop/exchange/server/node_modules/xhr2-cookies/dist/xml-http-request.js:252:61)\n    at ClientRequest.emit (events.js:198:13)\n    at Socket.socketErrorListener (_http_client.js:401:9)\n    at Socket.emit (events.js:198:13)\n    at emitErrorNT (internal/streams/destroy.js:91:8)\n    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)\n    at process._tickCallback (internal/process/next_tick.js:63:19)',
  level: 'error',
  timestamp: '2020-12-01T07:46:28.415Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:637:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T07:59:30.210Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:637:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:19:21.378Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:637:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:22:40.743Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:637:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:24:51.027Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:637:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:29:06.016Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:638:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:32:17.106Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:638:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:33:48.720Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:638:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T09:59:28.385Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:638:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T10:06:01.895Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:638:18\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T10:07:40.840Z' }
{ message: undefined,
  level: 'error',
  timestamp: '2020-12-01T10:55:15.506Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:682:19',
  level: 'error',
  timestamp: '2020-12-01T11:37:59.354Z' }
{ message:
   'TypeError: Cannot read property \'status\' of undefined\n    at /home/elliot/Desktop/exchange/server/routes/wallet.js:682:19\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  level: 'error',
  timestamp: '2020-12-01T11:50:53.158Z' }
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:12:25.060Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:17:36.116Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:18:20.159Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:19:14.895Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:19:26.937Z'
}
{
  message: undefined,
  level: 'error',
  timestamp: '2020-12-03T08:22:39.860Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:423:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at Function.process_params (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:410:3)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:174:3)\n' +
    '    at router (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:47:12)',
  level: 'error',
  timestamp: '2020-12-03T08:25:53.140Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:423:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at Function.process_params (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:410:3)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:174:3)\n' +
    '    at router (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:47:12)',
  level: 'error',
  timestamp: '2020-12-03T08:26:33.511Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:423:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at Function.process_params (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:410:3)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:174:3)\n' +
    '    at router (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:47:12)',
  level: 'error',
  timestamp: '2020-12-03T08:28:10.424Z'
}
{
  message: "Error: ENOENT: no such file or directory, open './images/coins/golden_bitcoin.png'\n" +
    '    at Object.openSync (fs.js:462:3)\n' +
    '    at Object.readFileSync (fs.js:364:35)\n' +
    '    at /home/hamid/projects/exchange/server/routes/service.js:423:24\n' +
    '    at Object.<anonymous> (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:47:42)\n' +
    '    at step (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:33:23)\n' +
    '    at Object.next (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:14:53)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:8:71\n' +
    '    at new Promise (<anonymous>)\n' +
    '    at __awaiter (/home/hamid/projects/exchange/server/middlewares/tryCtach.js:4:12)\n' +
    '    at /home/hamid/projects/exchange/server/middlewares/tryCtach.js:41:47\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:137:13)\n' +
    '    at Route.dispatch (/home/hamid/projects/exchange/server/node_modules/express/lib/router/route.js:112:3)\n' +
    '    at Layer.handle [as handle_request] (/home/hamid/projects/exchange/server/node_modules/express/lib/router/layer.js:95:5)\n' +
    '    at /home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:281:22\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:354:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at param (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:365:14)\n' +
    '    at Function.process_params (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:410:3)\n' +
    '    at next (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:275:10)\n' +
    '    at Function.handle (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:174:3)\n' +
    '    at router (/home/hamid/projects/exchange/server/node_modules/express/lib/router/index.js:47:12)',
  level: 'error',
  timestamp: '2020-12-03T08:28:45.669Z'
}
<<<<<<< HEAD
>>>>>>> 3c5d0ffa3c870a18254fc5d101dcee707be908b6
{
  message: 'getaddrinfo ENOTFOUND 9109769f43e4',
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    setName: 'rs0',
    maxSetVersion: 1,
    maxElectionId: 7fffffff0000000000000005,
    servers: Map(1) {
      '9109769f43e4:27017' => ServerDescription {
        address: '9109769f43e4:27017',
        error: Error: getaddrinfo ENOTFOUND 9109769f43e4
            at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
          name: 'MongoNetworkError'
        },
        roundTripTime: -1,
        lastUpdateTime: 155175990,
        lastWriteDate: null,
        opTime: null,
        type: 'Unknown',
        topologyVersion: undefined,
        minWireVersion: 0,
        maxWireVersion: 0,
        hosts: [],
        passives: [],
        arbiters: [],
        tags: []
      }
    },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: 9
  },
  level: 'error',
  timestamp: '2020-12-05T13:27:41.314Z'
}
{
  message: 'The find action on User Collection with email undefined has some errors: MongooseError: Operation `users.findOne()` buffering timed out after 10000ms',
  level: 'error',
  timestamp: '2020-12-05T13:27:51.158Z'
}
{
  message: 'getaddrinfo ENOTFOUND 9109769f43e4',
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    setName: 'rs0',
    maxSetVersion: 1,
    maxElectionId: 7fffffff0000000000000005,
    servers: Map(1) {
      '9109769f43e4:27017' => ServerDescription {
        address: '9109769f43e4:27017',
        error: Error: getaddrinfo ENOTFOUND 9109769f43e4
            at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
          name: 'MongoNetworkError'
        },
        roundTripTime: -1,
        lastUpdateTime: 155925743,
        lastWriteDate: null,
        opTime: null,
        type: 'Unknown',
        topologyVersion: undefined,
        minWireVersion: 0,
        maxWireVersion: 0,
        hosts: [],
        passives: [],
        arbiters: [],
        tags: []
      }
    },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: 9
  },
  level: 'error',
  timestamp: '2020-12-05T13:40:11.212Z'
}
=======
>>>>>>> 90aa51ca7ecfe01c90c76d80734326cf67570b30
{
  message: 'getaddrinfo ENOTFOUND 9109769f43e4',
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    setName: 'rs0',
    maxSetVersion: 1,
    maxElectionId: 7fffffff0000000000000005,
    servers: Map(1) {
      '9109769f43e4:27017' => ServerDescription {
        address: '9109769f43e4:27017',
        error: Error: getaddrinfo ENOTFOUND 9109769f43e4
            at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
          name: 'MongoNetworkError'
        },
        roundTripTime: -1,
        lastUpdateTime: 156197995,
        lastWriteDate: null,
        opTime: null,
        type: 'Unknown',
        topologyVersion: undefined,
        minWireVersion: 0,
        maxWireVersion: 0,
        hosts: [],
        passives: [],
        arbiters: [],
        tags: []
      }
    },
    stale: false,
    compatible: true,
    compatibilityError: null,
    logicalSessionTimeoutMinutes: null,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    commonWireVersion: 9
  },
  level: 'error',
  timestamp: '2020-12-05T13:44:43.435Z'
}
